dpnp/dpnp_iface_manipulation.py

```diff
@@ -496,25 +506,20 @@ def concatenate(
 
     """
 
+    if dtype is not None and out is not None:
+        raise TypeError(
+            "stack() only takes `out` or `dtype` as an argument, but both were provided."
+        )
+
+    usm_arrays = [dpnp.get_usm_ndarray(x) for x in arrays]
+    usm_res = dpt.concat(usm_arrays, axis=axis)
+    res = dpnp_array._create_from_usm_ndarray(usm_res)
+    if dtype is not None:
+        res = res.astype(dtype, casting=casting)
```

Comment:
we do not want to return a copy here, only when it is really necessary
```suggestion
        res = res.astype(dtype, casting=casting, copy=False)
```